index.js

```diff
@@ -369,6 +369,53 @@ module.exports = {
       })
     }
 
+    function excludeMembers(groupId, feedIds, opts = {}, cb) {
+      if (cb === undefined)
+        return promisify(excludeMembers)(groupId, feedIds, opts)
+
+      // TODO: should probably check this against a spec of its own
+      // TODO: should add members tangle to this. should we add opts to publish()? { spec, tangles, keys}
+      publish(
+        {
+          type: 'group/exclude',
+          excludes: feedIds,
+          recps: [groupId],
+        },
+        (err, exclusionMsg) => {
+          // prettier-ignore
+          if (err) return cb(clarify(err, 'Failed to publish exclude msg'))
+
+          pull(
+            listMembers(groupId),
+            pull.collect((err, beforeMembers) => {
+              const remainingMembers = beforeMembers.filter(
+                (member) => !feedIds.includes(member)
+              )
+
+              // TODO: add this key to ourselves. step 1. make sure we ourselves can post to the new feed. step 2 and later: other people can post on the new key
+              const newGroupKey = new SecretKey()
+              const newKeyContent = {
+                type: 'group/move-epoch',
+                secret: newGroupKey.toString('base64'),
+                recps: [groupId, ...remainingMembers],
```

Comment:
This still has to be spec'd but I guess the recps would be `[epochId, ...remainingMembers]` (epochId needed so that members *added* to this epoch will be able to decrypt this msg).

Reply:
I think we decided that these messages were gonna be published on the old feed? They won't be able to find the new feed if the message is published on the new feed itself.

Reply:
Hmm you're correct, drop what I said about `epochId`. They're published on the old feed, BUT they should not use the groupId for box2 encryption (this would mean that the Oscar would know the new epoch key).

So it would be `recps: remainingMembers` only.

Reply:
oh yeah lol good point

Reply:
hmm since the other members should be able to read the msg it should at least be encrypted to the new key